.github/workflows/reusable-run-go-acceptance-test.yml

```diff
@@ -0,0 +1,42 @@
+name: Run Reusable Go Acceptance Test
+
+on:
+  workflow_call:
+    inputs:
+      name:
+        description: 'The name to use that will appear in the output log file artifact'
+        required: true
+        type: string
+      path:
+        description: 'The path to the test without the precedeing "./" or following "/..." e.g. go test -v ./$path/...'
+        required: true
+        type: string
+    # We will need to add the capacity for receiving passed secrets once we get to the tests that require API credentials
+
+env:
+  VAULT_ACC: 1
+
+jobs:
+  product-metadata:
+    runs-on: ubuntu-latest
+    outputs:
+      go-version: ${{ steps.get-metadata.outputs.go-version }}
+    steps:
+      - uses: actions/checkout@v3
+      - id: get-metadata
+        run: echo "go-version=$(cat ./.go-version)" >> $GITHUB_OUTPUT
+
+  go-test:
+    runs-on: ubuntu-latest
+    needs: product-metadata
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ needs.product-metadata.outputs.go-version }}
+      - run: go test -v ./${{ inputs.path }}/... 2>&1 | tee ${{ inputs.name }}.txt
+      - uses: actions/upload-artifact@v3
+        with:
+          name: ${{ inputs.name }}-output
```

Comment:
Hrm. This is interesting. You can already get the output by downloading the log of workflow, but that isn't that great. This seems easier. We might want to put a shorter `retention` on this though, the 90 days seems a bit expensive for something we're going to be running nightly.